### PR TITLE
Support for "Next" pipelines

### PR DESCRIPTION
**Pull request by OwenArnold**
Optional publish docs (as needed for supporting `next` configurations).

See this in use [here](https://github.com/scipp/scippneutron/blob/next/.azure-pipelines/next.yml#L21) yielding this pipeline [next](https://dev.azure.com/scipp/scippneutron/_build?definitionId=56&_a=summary).

### CODE REVIEW

File: deploy.yml [hunk 1]
```diff
@@ -14,6 +14,9 @@ parameters:
 - name: gh_pages_subdirectory
   type: string
   default: ''
+- name: publish_docs
+  type: boolean
+  default: false
```
Comment (nvaytet) [inline deploy.yml:19]: By setting the default to `false`, you will prevent the `main` and `release` pipelines in `scipp` from publishing docs unless you also make changes there.
The easiest is probably to set it to `true` and then just have it as `false` in the next pipeline?

**Review by nvaytet — CHANGES_REQUESTED** [review]

File: deploy.yml (hunk 1 above)
Comment (OwenArnold) [reply deploy.yml:19]: I did for scippneutron, but I agree we want to avoid changing the others.

File: deploy.yml [hunk 2]
```diff
@@ -51,80 +54,81 @@ stages:
...
+          - ${{ if eq(parameters.publish_docs, true) }}:
+            - task: DownloadBuildArtifacts@0
+              inputs:
+                buildType: 'current'
+                specificBuildWithTriggering: true
+                downloadType: 'single'
+                artifactName: 'documentation'
+                downloadPath: '$(Build.StagingDirectory)'
+              displayName: 'Retrieve built documentation'
+          - ${{ if eq(parameters.publish_docs, true) }}:
```
Comment (nvaytet) [inline deploy.yml:70]: Why do you need a second if statement here? can't you put everything inside the same `if` block?

Comment (OwenArnold) [reply deploy.yml:70]: Not without significant rewrite I think? First task (anaconda upload) we still want. second two tasks may be disabled.

**Review by nvaytet — APPROVED** [review]

**Pull request merged**